tests/utils/test_flatdict.py

```diff
@@ -0,0 +1,24 @@
+from __future__ import annotations
+
+from poetry.utils.flatdict import flatten_dict
+
+
+def test_flatten_dict() -> None:
```

Comment:
since we are keeping the function, I would like to see more tests (different separators for example). Just because we only use `.` for now, doesn't mean we won't use the full capabilities of the function in the future.

Reply:
The test does use a non-default delimiter `:`